Fixes #16633 - AuthSourceLDAP uses *_authenticators filters

dLobatog opened the pull request:
Prior to this, non-admin users who were granted *_authenticators
permissions were not able to use them, as the controllers were looking
for *_auth_source_ldaps permissions instead.

dLobatog committed:
Fixes #16633 - AuthSourceLDAP uses *_authenticators filters

Prior to this, non-admin users who were granted *_authenticators
permissions were not able to use them, as the controllers were looking
for *_auth_source_ldaps permissions instead.

ares commented:
[test foreman]

ares commented on test/functional/auth_source_ldaps_controller_test.rb:
```diff
@@ -55,21 +55,27 @@ def test_destroy
-    users(:one).roles       = [Role.default, Role.find_by_name('Viewer')]
-  end
+  context 'user with viewer rights' do
+    # The 'Viewer' role has '*_authorizers' filters, so these tests test
```
nit: s/_authorizers/_authenticators/

ares commented on test/functional/auth_source_ldaps_controller_test.rb:
```diff
@@ -55,21 +55,27 @@ def test_destroy
-    setup_user
-    get :index
-    assert_response :success
+    test 'should succeed in viewing authentication source' do
```
This is quite useless test, index action returned 200 even before, the failure is only on granular filters but I know you didn't add it, so I leave it up to you if you want to remove it, improve it to test that user see some auth sources on page or just leave it as is.

ares reviewed (commented):
Works fine, just two nits. Good to merge then.

dLobatog commented:
@ares Thanks for reviewing, I've updated, this:
- removed the useless 'get index' test for the viewer role
- renamed the typo in my comment 'authorizers' to 'authenticators'

ares commented:
[test foreman] ACK pending jenkins

dLobatog commented:
@ares Should I rebase with develop and push? The integration tests are failing because this patch is based on 'develop' from a few days ago when it was broken

ares commented:
Thanks @dLobatog, merged as 8cc04d55c2cbe27a5c5fbb74254102f9d9658149
